depthai_ros_driver/src/depthai_loader.cpp

```diff
@@ -1,114 +1,52 @@
 #include <depthai_ros_driver/device_ros.hpp>
 #include <depthai_ros_driver/pipeline.hpp>
+#include <depthai_ros_driver/pipeline_loader.hpp>
 
 #include <depthai/pipeline/nodes.hpp>
 #include <depthai/openvino/OpenVINO.hpp>
 
 #include <pluginlib/class_loader.h>
 #include <ros/ros.h>
 
+// For generating pipeline config config
+#include <regex>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+
 int main(int argc, char** argv) {
     ros::init(argc, argv, "temp");
 
     ros::NodeHandle private_nh("~");
 
-    std::string pipeline_name = "rr::DefaultPipeline";
+    std::string pipeline_name = "depthai_ros_driver/MobilenetSSDPipeline";
     if (!private_nh.getParam("pipeline", pipeline_name)) {
         private_nh.setParam("pipeline", pipeline_name);
     }
 
-    std::string config;
-    if (!private_nh.getParam("config", config)) {
-        private_nh.setParam("config", config);
-    }
-
     std::string blob_file = "./mobilenet-ssd.blob";
-    private_nh.getParam("blob_file", blob_file);
-    auto openvino_version = dai::OpenVINO::Version::VERSION_2020_3;
-
-    // load pipeline using pluginlib
-    pluginlib::ClassLoader<rr::Pipeline> pipeline_loader("depthai_ros_driver", "rr::Pipeline");
-    try {
-    const auto loader = pipeline_loader.createUniqueInstance(pipeline_name);
-    loader->configure(config);
-    } catch (pluginlib::LibraryLoadException& ex) {
-        ROS_ERROR("The plugin failed to load for some reason. Error: %s", ex.what());
+    if (!private_nh.getParam("blob_file", blob_file)) {
+        private_nh.setParam("blob_file", blob_file);
```

Comment:
Should this be part of mobilenet ssd cpp file?

Reply:
True. and moving this into the pipeline cpp will remove config argument not necessary. I will fix it.